cube/src/main/java/zone/cogni/asquare/cube/shacl/ShaclToConversionProfile.java

```diff
@@ -131,20 +131,40 @@ private CompactConversionProfile.Attribute calculateAttribute(@Nonnull CompactCo
     attribute.setProperty(calculateProperty(propertyShape));
     attribute.setSingle(calculateSingle(propertyShape));
     attribute.setType(calculateAttributeType(propertyShape));
-    attribute.setInverse(false);
+    attribute.setInverse(calculateInverse(propertyShape));
     return attribute;
   }
 
+  private boolean calculateInverse(Resource propertyShape) {
+    if (propertyShape.hasProperty(Shacl.path)) return false;
+    if (propertyShape.hasProperty(Shacl.inversePath)) return true;
+
+    throw new RuntimeException("missing 'path' or 'inversePath' on property '" + propertyShape.getURI() + "'");
+  }
+
   @Nonnull
   private String calculatePathId(@Nonnull CompactConversionProfile profile,
                                  @Nonnull Resource propertyShape) {
-    Resource resource = propertyShape.getPropertyResourceValue(Shacl.path);
-    return convertResourceToId(profile, resource);
+    if (propertyShape.hasProperty(Shacl.path)) {
```

Comment:
shouldn't this be `if(calculateInverse(propertyShape)) {` ?

Reply:
makes sense. thinking of a new method since I typically use `calculate` as a one-shot.